sdk/core/core/src/az_http_policy_private.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_HTTP_POLICY_PRIVATE_H
+#define _az_HTTP_POLICY_PRIVATE_H
+
+#include <az_http.h>
+#include <az_result.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef AZ_NODISCARD az_result (*_az_http_pipeline_policy_fn)(
+    _az_http_policy* policies,
+    _az_http_request* ref_request,
+    az_http_response* ref_response);
+
+typedef struct
+{
+  _az_http_pipeline_policy_fn const func;
+  struct
+  {
+    _az_http_policy* policies;
+    _az_http_request* ref_request;
+    az_http_response* ref_response;
+  } params;
+} az_http_policy_callback;
```

Comment:
I don't think we should need this structure at all. From looking at your code, it appears that you inserted a level of indirection between a pipeline policy and a policy and because of this, you introduced this structure. I think if you moved the policy code into the pipeline policy function, you could get rid of this structure entirely and simplify the code. We can review in-person together

Reply:
Let's do it! We need the callback - if we want the code for retry policy and for the logging policy to be in different files other than all being in az_http_policy.c; otherise they won't be able to call the next policy.

Reply:
Done :)